packages/core/src/ObservableActorRef.ts

```diff
@@ -1,6 +1,5 @@
 import { EventObject, ActorRef, Behavior, SCXML, ActorContext } from './types';
 import { Actor } from './actor';
-import { CapturedState } from './capturedState';
 import { symbolObservable, toSCXMLEvent } from './utils';
 
 export class ObservableActorRef<TEvent extends EventObject, TEmitted>
```

Comment:
I think we need to establish the vocabulary here - we need to clarify what's:
- behavior
- actor
- actor ref
- observable actor ref
- spawned actor ref

+ we need to establish what's the key on the implementations object, is it `implementations.actors`? or `implementations.behaviors`?

Reply:
There should be only two concepts:

- behavior
- actor ref (created from a behavior)

Reply:
The interpreted machine can be a "system actor ref" which _can_ be stopped from the outside, and we can have child actors only stoppable by a special "stop" signal from the parent, so basically:

```ts
interface SystemRef extends ActorRef {
  stop: () => void; // stops the system
}
```

Info on stopping actors: https://doc.akka.io/docs/akka/current/typed/actor-lifecycle.html

Info on terminating an actor system: https://doc.akka.io/docs/akka/current/general/actor-systems.html#terminating-actorsystem